spec/unit/mutant/parser_spec.rb

```diff
@@ -6,16 +6,43 @@
   describe '#call' do
     let(:path) { instance_double(Pathname) }
 
+    let(:expected_node) { s(:sym, :source) }
+
     subject { object.call(path) }
 
+    let(:buffer) do
+      Unparser.buffer(<<~'RUBY')
+        :source # comment_a
+        :source # comment_b
+      RUBY
+    end
+
+    let(:expected_associations) do
+      associations = {}
+      associations.compare_by_identity
+
+      associations[s(:sym, :source)] = [
+        Parser::Source::Comment.new(Parser::Source::Range.new(buffer, 8, 19))
+      ]
+
+      associations[s(:sym, :source)] = [
+        Parser::Source::Comment.new(Parser::Source::Range.new(buffer, 28, 39))
+      ]
+
+      associations
+    end
+
     before do
-      expect(path).to receive(:read)
-        .with(no_args)
-        .and_return(':source')
+      allow(path).to receive(:read).and_return(buffer.source)
     end
 
     it 'returns parsed source' do
-      expect(subject).to eql(s(:sym, :source))
+      expect(subject.inspect).to eql(
```

Comment:
Comparing via inspect as `Parser::Source::*` only does identity on `#==` and `#eql?`. And writing peeking assertions is not as exhaustive. Basically an in-memory golden test :P